File: src/main/scala/org/openapitools/client/api/SdtmClassVariable.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import SdtmClassVariable._

case class SdtmClassVariable (
  ordinal: Option[String],
name: Option[String],
label: Option[String],
description: Option[String],
role: Option[String],
roleDescription: Option[String],
simpleDatatype: Option[String],
describedValueDomain: Option[String],
links: Option[SdtmClassVariableLinks])

object SdtmClassVariable {
  import DateTimeCodecs._

  implicit val SdtmClassVariableCodecJson: CodecJson[SdtmClassVariable] = CodecJson.derive[SdtmClassVariable]
  implicit val SdtmClassVariableDecoder: EntityDecoder[SdtmClassVariable] = jsonOf[SdtmClassVariable]
  implicit val SdtmClassVariableEncoder: EntityEncoder[SdtmClassVariable] = jsonEncoderOf[SdtmClassVariable]
}

File: client/clojure/src/cdisc_library_api/specs/sdtm_class_variable.clj
(ns cdisc-library-api.specs.sdtm-class-variable
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.sdtm-class-variable-links :refer :all]
            )
  (:import (java.io File)))


(def sdtm-class-variable-data
  {
   (ds/opt :ordinal) string?
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :role) string?
   (ds/opt :roleDescription) string?
   (ds/opt :simpleDatatype) string?
   (ds/opt :describedValueDomain) string?
   (ds/opt :_links) sdtm-class-variable-links-spec
   })

(def sdtm-class-variable-spec
  (ds/spec
    {:name ::sdtm-class-variable
     :spec sdtm-class-variable-data}))
